Tests/ApolloCacheDependentTests/WatchQueryTests.swift

```diff
@@ -4,700 +4,986 @@ import ApolloTestSupport
+            ]
+          ]
         ]
-      ],
-      "QUERY_ROOT.hero.friends.0": ["__typename": "Human", "name": "Luke Skywalker"],
-      "QUERY_ROOT.hero.friends.1": ["__typename": "Human", "name": "Han Solo"],
-      "QUERY_ROOT.hero.friends.2": ["__typename": "Human", "name": "Leia Organa"],
-      ]
-
-    withCache(initialRecords: initialRecords) { cache in
-      let store = ApolloStore(cache: cache)
-      let networkTransport = MockNetworkTransport(body: [
-        "data": [
-          "hero": [
-            "name": "Artoo",
-            "__typename": "Droid"
+      }
+      
+      let initialResultExpectation = resultObserver.expectation(description: "Watcher received initial result from server") { result in
+        let graphQLResult = try result.get()
+        XCTAssertEqual(graphQLResult.source, .server)
+        XCTAssertNil(graphQLResult.errors)
+        
+        let data = try XCTUnwrap(graphQLResult.data)
+        XCTAssertEqual(data.hero?.name, "R2-D2")
+      }
+      
+      watcher.fetch(cachePolicy: .fetchIgnoringCacheData)
+      
+      wait(for: [serverExpectation, initialResultExpectation], timeout: 1)
+    }
+    
+    runActivity("Fetch same query from server with different argument") { _ in
+      let serverExpectation = server.expect(HeroNameQuery.self) { request in
+        XCTAssertEqual(request.operation.episode, .jedi)
+        
+        return [
+          "data": [
+            "hero": [
+              "name": "Artoo",
+              "__typename": "Droid"
+            ]
           ]
         ]
-      ], store: store)
-      let client = ApolloClient(networkTransport: networkTransport, store: store)
-
-      let query = HeroAndFriendsNamesQuery()
-
-      var verifyResult: GraphQLResultHandler<HeroAndFriendsNamesQuery.Data>
-
-      verifyResult = { result in
-        switch result {
-        case .success(let graphQLResult):
-          XCTAssertNil(graphQLResult.errors)
-          guard let data = graphQLResult.data else {
-            XCTFail("No data in graphQLResult!")
-            return
-          }
-          
-          XCTAssertEqual(data.hero?.name, "R2-D2")
-          let friendsNames = data.hero?.friends?.compactMap { $0?.name }
-          XCTAssertEqual(friendsNames, ["Luke Skywalker", "Han Solo", "Leia Organa"])
-        case .failure(let error):
-          XCTFail("Unexpected error: \(error)")
-        }
       }
-
-      var expectation = self.expectation(description: "Fetching query")
-
-      _ = client.watch(query: query) { result in
-        verifyResult(result)
-        expectation.fulfill()
+      
+      let noUpdatedResultExpectation = resultObserver.expectation(description: "Other query shouldn't trigger refetch")
+      noUpdatedResultExpectation.isInverted = true
+      
+      let fetchedResultExpectation = fetch(query: HeroNameQuery(episode: .jedi), cachePolicy: .fetchIgnoringCacheData)
+        .expectation(description: "Fetched same query with different argument")
```

Comment:
Hmm, but that wouldn't allow you to set up multiple expectations or configure the observer in other ways (if we ever need that). We could add more arguments to `fetchWithExpectation` to support those uses, but at that point we'd be duplicating the observer's API. My thinking was that returning a handle like this would feel similar to other expectation-based testing APIs.

Reply:
> Hmm, but that wouldn't allow you to set up multiple expectations or configure the observer in other ways (if we ever need that)

Can you give me an example of when multiple expectations would be necessary? I definitely want to keep things forward thinking but I also don't want to solve literally every problem that could ever occur 😃

> other expectation-based testing APIs

Are you thinking of anything other than `XCTestCase` here?

Reply:
> Can you give me an example of when multiple expectations would be necessary? I definitely want to keep things forward thinking but I also don't want to solve literally every problem that could ever occur 😃

Multiple expectations for fetches are pretty common. You would first want to test for a result from the cache and then for a result from the server for example.

> > other expectation-based testing APIs
> Are you thinking of anything other than `XCTestCase` here?

Yeah, I was thinking of BDD-style testing frameworks like Jest or RSpec that use `expect` to return an object on which you call methods to configure assertions.

Reply:
Makes sense on multiple test expectations for cache vs network.

I personally am not a giant fan of Rspec style testing, I find it way harder to understand. But I think the thing with being able to create multiple expectations off the same fetch is a good enough reason to keep this as is.